beacon_node/beacon_processor/src/lib.rs

```diff
@@ -1583,10 +1208,7 @@ impl<T: BeaconChainTypes> BeaconProcessor<T> {
     /// Spawns a blocking worker thread to process some `Work`.
     ///
     /// Sends an message on `idle_tx` when the work is complete and the task is stopping.
-    fn spawn_worker(&mut self, work: Work<T>, toolbox: Toolbox<T>) {
-        let idle_tx = toolbox.idle_tx;
-        let work_reprocessing_tx = toolbox.work_reprocessing_tx;
-
+    fn spawn_worker(&mut self, work: Work<E>, idle_tx: mpsc::Sender<()>) {
```

Comment:
This function is largely simplified, rather than mapping each `Work` variant to a specific function, it simply runs the functions that are already contained in the `Work` variant.